include/h2o/socketpool.h

```diff
@@ -67,11 +81,20 @@ typedef struct st_h2o_socketpool_t {
         pthread_mutex_t mutex;
         h2o_linklist_t sockets; /* guarded by the mutex; list of struct pool_entry_t defined in socket/pool.c */
     } _shared;
+
+    /* vars used by load balancing, modified by multiple threads */
+    struct {
+        h2o_socketpool_lb_selector selector;
+        h2o_socketpool_lb_dispose_cb dispose;
+        pthread_mutex_t mutex;
```

Comment:
since the mutex might or might not be needed by the load balancing policy, it might make sense to move it to `struct round_robin_t` and have it locked in `lb_selector`?